### PR TITLE
Get Azuracast baseline - Fix port reassignment - WIP - needs testing

### PR DESCRIPTION
**Pull request by jvonau**
not for merging

### CODE REVIEW

**Comment by jvonau** [comment]
@holta Can you run the PR so we can get some better information for #2943?

**Comment by holta** [comment]
> @holta Can you run the PR so we can get some better information for #2943?

It's running on 10.8.0.54 = 192-u20-desk-2946

Feel free to log in as nec.

cc: @m-anish

**Comment by jvonau** [comment]
http://sprunge.us/XQI7ti

**Comment by jvonau** [comment]
new wrinkle with FF when trying http://10.8.0.54:10080/
>This address is restricted
This address uses a network port which is normally used for purposes other than Web browsing. Firefox has canceled the request for your protection.

Fix FF and I see nginx 502 bad gateway. Shutdown down the main nginx service and the same error is present suggesting the container's nginx is still operational and the iptables rules work.

**Comment by jvonau** [comment]
```
CONTAINER ID   NAME                 CPU %     MEM USAGE / LIMIT     MEM %     NET I/O           BLOCK I/O        PIDS
2243598e3ee5   nginx_proxy          0.75%     66.36MiB / 1.939GiB   3.34%     1.64kB / 1.36kB   1.74MB / 807kB   19
ec083379f789   azuracast_web        0.53%     7.785MiB / 1.939GiB   0.39%     48.9kB / 67.5kB   0B / 0B          2
382ae8157e21   azuracast_mariadb    0.00%     0B / 0B               0.00%     0B / 0B           0B / 0B          0
d1716c14d07d   azuracast_stations   0.24%     22.98MiB / 1.939GiB   1.16%     42.6kB / 0B       2.29MB / 102kB   8
a2c37ce47be8   azuracast_redis      0.57%     2.066MiB / 1.939GiB   0.10%     6.23MB / 4.49MB   188kB / 0B       5
```
```
sudo ls /run/docker/containerd
2243598e3ee583bb39fffb5581e32b3e41fc9a84c41fef4417a6f7857606093c  d1716c14d07d77eabca45e282c85d23cd3d392d976db10dc9d7880fe4aa7ceaa
a2c37ce47be850305d6607c26289443d201bc48257da78a511b2f4141546d22c  ec083379f789395bfdbacbd3611beef48eb6530572a811bceca4ad21ecebf9db
```
```
iiab-admin@box:~$ sudo docker network ls
NETWORK ID     NAME                 DRIVER    SCOPE
5b0455813641   azuracast_backend    bridge    local
7f81622c243b   azuracast_default    bridge    local
2080a308961d   azuracast_frontend   bridge    local
74cd419f3b5d   bridge               bridge    local
04e1afbcf4cf   host                 host      local
9c9c74d58544   none                 null      local
```
```
sudo docker network inspect 5b0455813641
[
    {
        "Name": "azuracast_backend",
        "Id": "5b0455813641db78a7f7d71565ba001dee4b94d0219c93f500f35d8f3cb84304",
        "Created": "2021-08-14T14:06:44.410920741-04:00",
        "Scope": "local",
        "Driver": "bridge",
        "EnableIPv6": false,
        "IPAM": {
            "Driver": "default",
            "Options": null,
            "Config": [
                {
                    "Subnet": "172.19.0.0/16",
                    "Gateway": "172.19.0.1"
                }
            ]
        },
        "Internal": false,
        "Attachable": true,
        "Ingress": false,
        "ConfigFrom": {
            "Network": ""
        },
        "ConfigOnly": false,
        "Containers": {
            "a2c37ce47be850305d6607c26289443d201bc48257da78a511b2f4141546d22c": {
                "Name": "azuracast_redis",
                "EndpointID": "899850bc8e19cbe884f96471d35ba805feff1710e4b21f869c8c613d15a95fa0",
                "MacAddress": "02:42:ac:13:00:02",
                "IPv4Address": "172.19.0.2/16",
                "IPv6Address": ""
            },
            "d1716c14d07d77eabca45e282c85d23cd3d392d976db10dc9d7880fe4aa7ceaa": {
                "Name": "azuracast_stations",
                "EndpointID": "7243819a84d70435dde5b860d4bbc36cff1f1e118d75af0e86a1f986a36958bc",
                "MacAddress": "02:42:ac:13:00:04",
                "IPv4Address": "172.19.0.4/16",
                "IPv6Address": ""
            },
            "ec083379f789395bfdbacbd3611beef48eb6530572a811bceca4ad21ecebf9db": {
                "Name": "azuracast_web",
                "EndpointID": "98167edc45f8eb12de8d34026b4017881167a3591ef7464576166a635c178e88",
                "MacAddress": "02:42:ac:13:00:03",
                "IPv4Address": "172.19.0.3/16",
                "IPv6Address": ""
            }
        },
        "Options": {},
        "Labels": {
            "com.docker.compose.network": "backend",
            "com.docker.compose.project": "azuracast",
            "com.docker.compose.version": "1.29.2"
        }
    }
]
```
```
iiab-admin@box:~$ sudo docker network inspect 7f81622c243b
[
    {
        "Name": "azuracast_default",
        "Id": "7f81622c243baf31c016757406c4f633d3119a2d7a63c07fac855de919d6bbd0",
        "Created": "2021-08-14T14:05:34.745884117-04:00",
        "Scope": "local",
        "Driver": "bridge",
        "EnableIPv6": false,
        "IPAM": {
            "Driver": "default",
            "Options": null,
            "Config": [
                {
                    "Subnet": "172.18.0.0/16",
                    "Gateway": "172.18.0.1"
                }
            ]
        },
        "Internal": false,
        "Attachable": true,
        "Ingress": false,
        "ConfigFrom": {
            "Network": ""
        },
        "ConfigOnly": false,
        "Containers": {},
        "Options": {},
        "Labels": {
            "com.docker.compose.network": "default",
            "com.docker.compose.project": "azuracast",
            "com.docker.compose.version": "1.29.2"
        }
    }
]
```
```
iiab-admin@box:~$ sudo docker network inspect 2080a308961d
[
    {
        "Name": "azuracast_frontend",
        "Id": "2080a308961d65cb83eff411b1f60cf300da501ec4c252b70ee8f852af9ebeb5",
        "Created": "2021-08-14T14:06:44.695410645-04:00",
        "Scope": "local",
        "Driver": "bridge",
        "EnableIPv6": false,
        "IPAM": {
            "Driver": "default",
            "Options": null,
            "Config": [
                {
                    "Subnet": "172.20.0.0/16",
                    "Gateway": "172.20.0.1"
                }
            ]
        },
        "Internal": false,
        "Attachable": true,
        "Ingress": false,
        "ConfigFrom": {
            "Network": ""
        },
        "ConfigOnly": false,
        "Containers": {
            "2243598e3ee583bb39fffb5581e32b3e41fc9a84c41fef4417a6f7857606093c": {
                "Name": "nginx_proxy",
                "EndpointID": "7cf4a945bcd02086fd91c76777fbf30bf776c5fb42b070e28b8639e4ce845681",
                "MacAddress": "02:42:ac:14:00:04",
                "IPv4Address": "172.20.0.4/16",
                "IPv6Address": ""
            },
            "d1716c14d07d77eabca45e282c85d23cd3d392d976db10dc9d7880fe4aa7ceaa": {
                "Name": "azuracast_stations",
                "EndpointID": "e9d531c4affd03e7895d2c6d177dee95cac9a761db3b55326a73a566761fe84e",
                "MacAddress": "02:42:ac:14:00:02",
                "IPv4Address": "172.20.0.2/16",
                "IPv6Address": ""
            },
            "ec083379f789395bfdbacbd3611beef48eb6530572a811bceca4ad21ecebf9db": {
                "Name": "azuracast_web",
                "EndpointID": "88d559d2bcafaf68fbcf9bd19ea2a8ef26e622da94862f69ef57c5c625873b4d",
                "MacAddress": "02:42:ac:14:00:03",
                "IPv4Address": "172.20.0.3/16",
                "IPv6Address": ""
            }
        },
        "Options": {},
        "Labels": {
            "com.docker.compose.network": "frontend",
            "com.docker.compose.project": "azuracast",
            "com.docker.compose.version": "1.29.2"
        }
    }
]
```
```
iiab-admin@box:~$ sudo docker network inspect 74cd419f3b5d
[
    {
        "Name": "bridge",
        "Id": "74cd419f3b5d0b0c40f04b5b29c7aa0f0cb4708810c8f83c612b3f2965ac7c03",
        "Created": "2021-08-14T14:04:37.383916612-04:00",
        "Scope": "local",
        "Driver": "bridge",
        "EnableIPv6": false,
        "IPAM": {
            "Driver": "default",
            "Options": null,
            "Config": [
                {
                    "Subnet": "172.17.0.0/16"
                }
            ]
        },
        "Internal": false,
        "Attachable": false,
        "Ingress": false,
        "ConfigFrom": {
            "Network": ""
        },
        "ConfigOnly": false,
        "Containers": {},
        "Options": {
            "com.docker.network.bridge.default_bridge": "true",
            "com.docker.network.bridge.enable_icc": "true",
            "com.docker.network.bridge.enable_ip_masquerade": "true",
            "com.docker.network.bridge.host_binding_ipv4": "0.0.0.0",
            "com.docker.network.bridge.name": "docker0",
            "com.docker.network.driver.mtu": "1500"
        },
        "Labels": {}
    }
]
```
```
iiab-admin@box:~$ sudo docker network inspect 04e1afbcf4cf
[
    {
        "Name": "host",
        "Id": "04e1afbcf4cf87f09dd40ea0218beedf585da17fb0af1459c3fa19b1ec84b323",
        "Created": "2021-08-14T14:04:37.318952413-04:00",
        "Scope": "local",
        "Driver": "host",
        "EnableIPv6": false,
        "IPAM": {
            "Driver": "default",
            "Options": null,
            "Config": []
        },
        "Internal": false,
        "Attachable": false,
        "Ingress": false,
        "ConfigFrom": {
            "Network": ""
        },
        "ConfigOnly": false,
        "Containers": {},
        "Options": {},
        "Labels": {}
    }
]
```
```
iiab-admin@box:~$ sudo docker network inspect 9c9c74d58544
[
    {
        "Name": "none",
        "Id": "9c9c74d58544c7444c8e9d9b555f2ffeb941c8914ea334e3f75940a41587db52",
        "Created": "2021-08-14T14:04:37.266864242-04:00",
        "Scope": "local",
        "Driver": "null",
        "EnableIPv6": false,
        "IPAM": {
            "Driver": "default",
            "Options": null,
            "Config": []
        },
        "Internal": false,
        "Attachable": false,
        "Ingress": false,
        "ConfigFrom": {
            "Network": ""
        },
        "ConfigOnly": false,
        "Containers": {},
        "Options": {},
        "Labels": {}
    }
]
```

**Comment by jvonau** [comment]
```
iiab-admin@box:/opt/azuracast$ cat docker-compose.yml 
version: '2.2'
services:
    nginx_proxy:
        container_name: nginx_proxy
        image: 'ghcr.io/azuracast/nginx_proxy:${AZURACAST_VERSION:-latest}'
        ports:
            - '${AZURACAST_HTTP_PORT:-80}:80'
            - '${AZURACAST_HTTPS_PORT:-443}:443'
        volumes:
            - 'letsencrypt:/etc/nginx/certs'
            - 'nginx_proxy_vhosts:/etc/nginx/vhost.d'
            - 'letsencrypt_html:/usr/share/nginx/html'
            - '/var/run/docker.sock:/tmp/docker.sock:ro'
        environment:
            NGINX_TIMEOUT: '${NGINX_TIMEOUT:-1800}'
            DEFAULT_HOST: '${LETSENCRYPT_HOST:-azuracast.local}'
        networks:
            - frontend
        depends_on:
            - web
        restart: always
    web:
        container_name: azuracast_web
        image: 'ghcr.io/azuracast/web:${AZURACAST_VERSION:-latest}'
        ports:
            - '${AZURACAST_SFTP_PORT:-2022}:2022'
        depends_on:
            - mariadb
            - stations
            - redis
        env_file: azuracast.env
        environment:
            LANG: '${LANG:-en_US.UTF-8}'
            AZURACAST_DC_REVISION: 12
            AZURACAST_VERSION: '${AZURACAST_VERSION:-latest}'
            AZURACAST_SFTP_PORT: '${AZURACAST_SFTP_PORT:-2022}'
            VIRTUAL_HOST: '${LETSENCRYPT_HOST:-azuracast.local}'
            NGINX_TIMEOUT: '${NGINX_TIMEOUT:-1800}'
            PUID: '${AZURACAST_PUID:-1000}'
            PGID: '${AZURACAST_PGID:-1000}'
        volumes:
            - 'letsencrypt:/etc/nginx/certs:ro'
            - 'www_vendor:/var/azuracast/www/vendor'
            - 'www_uploads:/var/azuracast/uploads'
            - 'tmp_data:/var/azuracast/www_tmp'
            - 'station_data:/var/azuracast/stations'
            - 'shoutcast2_install:/var/azuracast/servers/shoutcast2'
            - 'geolite_install:/var/azuracast/geoip'
            - 'sftpgo_data:/var/azuracast/sftpgo/persist'
            - 'backups:/var/azuracast/backups'
        networks:
            - frontend
            - backend
        restart: always
        ulimits:
            nofile:
                soft: 65536
                hard: 65536
        logging:
            options:
                max-size: 1m
                max-file: '5'
    mariadb:
        container_name: azuracast_mariadb
        image: 'ghcr.io/azuracast/db:${AZURACAST_VERSION:-latest}'
        volumes:
            - 'db_data:/var/lib/mysql'
        env_file: azuracast.env
        networks:
            - backend
        restart: always
        logging:
            options:
                max-size: 1m
                max-file: '5'
    redis:
        container_name: azuracast_redis
        image: 'ghcr.io/azuracast/redis:${AZURACAST_VERSION:-latest}'
        sysctls:
            net.core.somaxconn: 1024
        volumes:
            - 'redis_data:/data'
        networks:
            - backend
        restart: always
        logging:
            options:
                max-size: 1m
                max-file: '5'
    stations:
        container_name: azuracast_stations
        image: 'ghcr.io/azuracast/radio:${AZURACAST_VERSION:-latest}'
        environment:
            PUID: '${AZURACAST_PUID:-1000}'
            PGID: '${AZURACAST_PGID:-1000}'
        ports:
            - '8000:8000'
            - '8005:8005'
            - '8006:8006'
            - '8010:8010'
            - '8015:8015'
            - '8016:8016'
            - '8020:8020'
            - '8025:8025'
            - '8026:8026'
            - '8030:8030'
            - '8035:8035'
            - '8036:8036'
            - '8040:8040'
            - '8045:8045'
            - '8046:8046'
            - '8050:8050'
            - '8055:8055'
            - '8056:8056'
            - '8060:8060'
            - '8065:8065'
            - '8066:8066'
            - '8070:8070'
            - '8075:8075'
            - '8076:8076'
            - '8090:8090'
            - '8095:8095'
            - '8096:8096'
            - '8100:8100'
            - '8105:8105'
            - '8106:8106'
            - '8110:8110'
            - '8115:8115'
            - '8116:8116'
            - '8120:8120'
            - '8125:8125'
            - '8126:8126'
            - '8130:8130'
            - '8135:8135'
            - '8136:8136'
            - '8140:8140'
            - '8145:8145'
            - '8146:8146'
            - '8150:8150'
            - '8155:8155'
            - '8156:8156'
            - '8160:8160'
            - '8165:8165'
            - '8166:8166'
            - '8170:8170'
            - '8175:8175'
            - '8176:8176'
            - '8180:8180'
            - '8185:8185'
            - '8186:8186'
            - '8190:8190'
            - '8195:8195'
            - '8196:8196'
            - '8200:8200'
            - '8205:8205'
            - '8206:8206'
            - '8210:8210'
            - '8215:8215'
            - '8216:8216'
            - '8220:8220'
            - '8225:8225'
            - '8226:8226'
            - '8230:8230'
            - '8235:8235'
            - '8236:8236'
            - '8240:8240'
            - '8245:8245'
            - '8246:8246'
            - '8250:8250'
            - '8255:8255'
            - '8256:8256'
            - '8260:8260'
            - '8265:8265'
            - '8266:8266'
            - '8270:8270'
            - '8275:8275'
            - '8276:8276'
            - '8280:8280'
            - '8285:8285'
            - '8286:8286'
            - '8290:8290'
            - '8295:8295'
            - '8296:8296'
            - '8300:8300'
            - '8305:8305'
            - '8306:8306'
            - '8310:8310'
            - '8315:8315'
            - '8316:8316'
            - '8320:8320'
            - '8325:8325'
            - '8326:8326'
            - '8330:8330'
            - '8335:8335'
            - '8336:8336'
            - '8340:8340'
            - '8345:8345'
            - '8346:8346'
            - '8350:8350'
            - '8355:8355'
            - '8356:8356'
            - '8360:8360'
            - '8365:8365'
            - '8366:8366'
            - '8370:8370'
            - '8375:8375'
            - '8376:8376'
            - '8380:8380'
            - '8385:8385'
            - '8386:8386'
            - '8390:8390'
            - '8395:8395'
            - '8396:8396'
            - '8400:8400'
            - '8405:8405'
            - '8406:8406'
            - '8410:8410'
            - '8415:8415'
            - '8416:8416'
            - '8420:8420'
            - '8425:8425'
            - '8426:8426'
            - '8430:8430'
            - '8435:8435'
            - '8436:8436'
            - '8440:8440'
            - '8445:8445'
            - '8446:8446'
            - '8450:8450'
            - '8455:8455'
            - '8456:8456'
            - '8460:8460'
            - '8465:8465'
            - '8466:8466'
            - '8470:8470'
            - '8475:8475'
            - '8476:8476'
            - '8480:8480'
            - '8485:8485'
            - '8486:8486'
            - '8490:8490'
            - '8495:8495'
            - '8496:8496'
        volumes:
            - 'station_data:/var/azuracast/stations'
            - 'shoutcast2_install:/var/azuracast/servers/shoutcast2'
            - 'letsencrypt:/etc/nginx/certs'
            - 'tmp_data:/var/azuracast/www_tmp'
        networks:
            - frontend
            - backend
        init: true
        restart: always
        ulimits:
            nofile:
                soft: 65536
                hard: 65536
        logging:
            options:
                max-size: 1m
                max-file: '5'
networks:
    frontend:
        driver: bridge
    backend:
        driver: bridge
volumes:
    nginx_proxy_vhosts: {  }
    db_data: {  }
    letsencrypt: {  }
    letsencrypt_html: {  }
    letsencrypt_acme: {  }
    shoutcast2_install: {  }
    geolite_install: {  }
    sftpgo_data: {  }
    station_data: {  }
    www_vendor: {  }
    www_uploads: {  }
    tmp_data: {  }
    redis_data: {  }
    backups: {  }
```
```
iiab-admin@box:/opt/azuracast$ cat azuracast.env 
# This file was automatically generated by AzuraCast.
# You can modify it as necessary. To apply changes, restart the Docker containers.
# Remove the leading "#" symbol from lines to uncomment them.

# The locale to use for CLI commands.
# Valid options: en_US, cs_CZ, de_DE, es_ES, fr_FR, el_GR, it_IT, hu_HU, nl_NL, pl_PL, pt_PT, pt_BR, ru_RU, sv_SE, tr_TR, zh_CN, ko_KR
# Default: en_US
LANG=en_US

# The application environment.
# Valid options: production, development, testing
# Default: production
APPLICATION_ENV=production

# Manually modify the logging level.
# This allows you to log debug-level errors temporarily (for problem-solving)
# or reduce the volume of logs that are produced by your installation,
# without needing to modify whether your installation is a production or
# development instance.
# Valid options: debug, info, notice, warning, error, critical, alert, emergency
# Default: notice
# LOG_LEVEL=notice

# Composer Plugin Mode
# Enable the composer "merge" functionality to combine the main application's
# composer.json file with any plugin composer files. This can have
# performance implications, so you should only use it if you use one or more
# plugins with their own Composer dependencies.
# Valid options: true, false
COMPOSER_PLUGIN_MODE=false

# Minimum Port for Station Port Assignment
# Modify this if your stations are listening on nonstandard ports.
# Default: 8000
# AUTO_ASSIGN_PORT_MIN=8000

# Maximum Port for Station Port Assignment
# Modify this if your stations are listening on nonstandard ports.
# Default: 8499
# AUTO_ASSIGN_PORT_MAX=8499

# MariaDB Host
# Do not modify this after installation.
# Default: mariadb
MYSQL_HOST=mariadb

# MariaDB Port
# Do not modify this after installation.
# Default: 3306
MYSQL_PORT=3306

# MariaDB Username
# Do not modify this after installation.
# Default: azuracast
MYSQL_USER=azuracast

# MariaDB Password
# Do not modify this after installation.
# Default: azur4c457
MYSQL_PASSWORD=FJX3jQedEyYe

# MariaDB Database Name
# Do not modify this after installation.
# Default: azuracast
MYSQL_DATABASE=azuracast

# Auto-generate Random MariaDB Root Password
# Do not modify this after installation.
# Default: yes
# MYSQL_RANDOM_ROOT_PASSWORD=yes

# Enable MariaDB Slow Query Log
# Log slower queries to diagnose possible database issues. Only turn this on
# if needed.
# MYSQL_SLOW_QUERY_LOG=

# MariaDB Maximum Connections
# Set the amount of allowed connections to the database. This value should be
# increased if you are seeing the "Too many connections" error in the logs.
# Default: 100
# MYSQL_MAX_CONNECTIONS=100

# Enable Redis
# Disable to use a flatfile cache instead of Redis.
# Default: true
# ENABLE_REDIS=true

# Redis Host
# Default: redis
REDIS_HOST=redis

# Redis Port
# Default: 6379
REDIS_PORT=6379

# Redis Database Index
# Valid options: 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15
# Default: 1
REDIS_DB=1

# PHP Maximum POST File Size
# Default: 25M
# PHP_MAX_FILE_SIZE=25M

# PHP Memory Limit
# Default: 128M
# PHP_MEMORY_LIMIT=128M

# PHP Script Maximum Execution Time
# (in seconds)
# Default: 30
# PHP_MAX_EXECUTION_TIME=30

# Short Sync Task Execution Time
# The maximum execution time (and lock timeout) for the 15-second, 1-minute
# and 5-minute synchronization tasks.
# Default: 600
# SYNC_SHORT_EXECUTION_TIME=600

# Long Sync Task Execution Time
# The maximum execution time (and lock timeout) for the 1-hour
# synchronization task.
# Default: 1800
# SYNC_LONG_EXECUTION_TIME=1800

# Maximum PHP-FPM Worker Processes
# Default: 5
# PHP_FPM_MAX_CHILDREN=5

# Enable Performance Profiling Extension
# Profiling data can be viewed by visiting
# http://your-azuracast-site/?SPX_KEY=dev&SPX_UI_URI=/.
# PROFILING_EXTENSION_ENABLED=

# Profile Performance on All Requests
# This will have a significant performance impact on your installation.
# PROFILING_EXTENSION_ALWAYS_ON=

# Profiling Extension HTTP Key
# The value for the "SPX_KEY" parameter for viewing profiling pages.
# Default: dev
# PROFILING_EXTENSION_HTTP_KEY=dev

# Profiling Extension IP Allow List
# Valid options: 127.0.0.1, *
# Default: *
# PROFILING_EXTENSION_HTTP_IP_WHITELIST=*
```
```
iiab-admin@box:/opt/azuracast$ cat .env 
# This file was automatically generated by AzuraCast.
# You can modify it as necessary. To apply changes, restart the Docker containers.
# Remove the leading "#" symbol from lines to uncomment them.

# (Docker Compose) All Docker containers are prefixed by this name. Do not change this after installation.
# Default: azuracast
COMPOSE_PROJECT_NAME=azuracast

# (Docker Compose) The amount of time to wait before a Docker Compose operation fails. Increase this on lower performance computers.
# Default: 300
COMPOSE_HTTP_TIMEOUT=300

# AzuraCast Release Channel
# Valid options: latest, stable
# Default: latest
AZURACAST_VERSION=stable

# HTTP Port
# The main port AzuraCast listens to for insecure HTTP connections.
# Default: 80
AZURACAST_HTTP_PORT=10080

# HTTPS Port
# The main port AzuraCast listens to for secure HTTPS connections.
# Default: 443
AZURACAST_HTTPS_PORT=10443

# SFTP Port
# The port AzuraCast listens to for SFTP file management connections.
# Default: 2022
# AZURACAST_SFTP_PORT=2022

# Station Ports
# The ports AzuraCast should listen to for station broadcasts and incoming DJ
# connections.
# Default: 8000,8005,8006,8010,8015,8016,8020,8025,8026,8030,8035,8036,8040,8045,8046,8050,8055,8056,8060,8065,8066,8070,8075,8076,8090,8095,8096,8100,8105,8106,8110,8115,8116,8120,8125,8126,8130,8135,8136,8140,8145,8146,8150,8155,8156,8160,8165,8166,8170,8175,8176,8180,8185,8186,8190,8195,8196,8200,8205,8206,8210,8215,8216,8220,8225,8226,8230,8235,8236,8240,8245,8246,8250,8255,8256,8260,8265,8266,8270,8275,8276,8280,8285,8286,8290,8295,8296,8300,8305,8306,8310,8315,8316,8320,8325,8326,8330,8335,8336,8340,8345,8346,8350,8355,8356,8360,8365,8366,8370,8375,8376,8380,8385,8386,8390,8395,8396,8400,8405,8406,8410,8415,8416,8420,8425,8426,8430,8435,8436,8440,8445,8446,8450,8455,8456,8460,8465,8466,8470,8475,8476,8480,8485,8486,8490,8495,8496
# AZURACAST_STATION_PORTS=8000,8005,8006,8010,8015,8016,8020,8025,8026,8030,8035,8036,8040,8045,8046,8050,8055,8056,8060,8065,8066,8070,8075,8076,8090,8095,8096,8100,8105,8106,8110,8115,8116,8120,8125,8126,8130,8135,8136,8140,8145,8146,8150,8155,8156,8160,8165,8166,8170,8175,8176,8180,8185,8186,8190,8195,8196,8200,8205,8206,8210,8215,8216,8220,8225,8226,8230,8235,8236,8240,8245,8246,8250,8255,8256,8260,8265,8266,8270,8275,8276,8280,8285,8286,8290,8295,8296,8300,8305,8306,8310,8315,8316,8320,8325,8326,8330,8335,8336,8340,8345,8346,8350,8355,8356,8360,8365,8366,8370,8375,8376,8380,8385,8386,8390,8395,8396,8400,8405,8406,8410,8415,8416,8420,8425,8426,8430,8435,8436,8440,8445,8446,8450,8455,8456,8460,8465,8466,8470,8475,8476,8480,8485,8486,8490,8495,8496

# Docker User UID
# Set the UID of the user running inside the Docker containers. Matching this
# with your host UID can fix permission issues.
# Default: 1000
# AZURACAST_PUID=1000

# Docker User GID
# Set the GID of the user running inside the Docker containers. Matching this
# with your host GID can fix permission issues.
# Default: 1000
# AZURACAST_PGID=1000

# Advanced: Use Privileged Docker Settings
# Default: true
# AZURACAST_COMPOSE_PRIVILEGED=true

# LetsEncrypt Domain Name(s)
# Domain name (example.com) or names (example.com,foo.bar) to use with
# LetsEncrypt.
# LETSENCRYPT_HOST=

# LetsEncrypt E-mail Address
# Optionally provide an e-mail address for updates from LetsEncrypt.
# LETSENCRYPT_EMAIL=
```

Don't think the ports were moved from 8XXX to 10XXX

**Comment by jvonau** [comment]
https://github.com/iiab/iiab/pull/2946#issuecomment-898942911
>Fix FF and I see nginx 502 bad gateway. Shutdown down the main nginx service and the same error is present suggesting the container's nginx is still operational and the iptables rules work.

@holta Ok lets take IIAB out of the picture can you spin up an VM with no roles enabled for a hand installation? mysql looks flaky to me the container doesn't stay running, basically going to do a ./docker.sh by hand..

**Comment by jvonau** [comment]
```
iiab-admin@box:/opt/azuracast$ sudo docker logs 1e6c205b976d
2021-08-15 05:41:07+00:00 [Note] [Entrypoint]: Entrypoint script for MariaDB Server 1:10.5.11+maria~focal started.
2021-08-15 05:41:07+00:00 [Note] [Entrypoint]: Switching to dedicated user 'mysql'
2021-08-15 05:41:07+00:00 [Note] [Entrypoint]: Entrypoint script for MariaDB Server 1:10.5.11+maria~focal started.
2021-08-15 05:41:07+00:00 [ERROR] [Entrypoint]: Database is uninitialized and password option is not specified
	You need to specify one of MARIADB_ROOT_PASSWORD, MARIADB_ALLOW_EMPTY_ROOT_PASSWORD and MARIADB_RANDOM_ROOT_PASSWORD
```
yup mysql is flaky

**Comment by jvonau** [comment]
Compose file and env files need tinkering http://10.8.0.54:10088 Have a test spin @holta

**Comment by jvonau** [comment]
https://github.com/AzuraCast/AzuraCast/issues/4491

**Comment by jvonau** [comment]
@holta well upstream says its fix and `./docker.sh upgrade` pulled down a new mariadb container today where as yesterday it didn't. I would say retest, at least the networking is working currently at the above url without the override file in place simplifying the layout. I noted that the ports were not moved so something is still amiss.

**Comment by jvonau** [comment]
Think commit e7c8599 should address the ports not being altered. I'm tight for space so I can't test this locally.

**Comment by jvonau** [comment]
Could retrieve azuracast.env then alter the ports with lineinfile for future proofing but for today this is enough to test with and there is a sample for reference.

**Comment by jvonau** [comment]
0871592ed61704f9cb95048bd6ebf8760ebdf6be

**Comment by holta** [comment]
FYI @jvonau and I will look closely at this PR in coming days if @m-anish's friends in Ladakh in fact do now need this.

As appears quite likely.  But we'll know more in coming days.

**Comment by holta** [comment]
At the moment there are 10 open tickets that mention AzuraCast: (1 PR + 9 issues)
https://github.com/iiab/iiab/search?q=azuracast&state=open&type=issues

Here are 3 that stand out:
- #1770
- #2943
- #2991

**Comment by jvonau** [comment]
https://github.com/AzuraCast/AzuraCast/issues/5191

File: roles/azuracast/upstream/docker-compose.sample.yml [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+    # https://docs.azuracast.com/en/administration/docker#using-non-standard-ports
+    ports:
+      - '${AZURACAST_HTTP_PORT:-80}:80'
+      - '${AZURACAST_HTTPS_PORT:-443}:443'
+      - '${AZURACAST_SFTP_PORT:-2022}:2022'
+      # This default mapping is the outgoing and incoming ports for the first 50 stations.
+      # You can override this port mapping in your own docker-compose.override.yml file.
+      # For instructions, see:
+      # https://docs.azuracast.com/en/administration/docker#expanding-the-station-port-range
+      - '8000:8000'
```
Comment (holta) [inline roles/azuracast/upstream/docker-compose.sample.yml:27]: What's the purpose all these ~150 lines?

FWIW quite a number of these overlap (conflict?) with other IIAB ports in the 8000-to-8500 range.

Should any adjustments and/or clarifications be made &mdash; either here or at https://github.com/iiab/iiab/wiki/IIAB-Networking#list-of-ports--services ?

**Comment by holta** [comment]
This PR was installed successfully onto an existing IIAB on a RPi 4 "183-lite64-SMALL+azuracast" here:

10.8.0.46

@jvonau can you test/confirm that AzuraCast is working there as it should?
(Hopefully 2GB RAM is enough for AzuraCast's Docker needs!)

Historical / Original PR from @m-anish, just FYI for those interested:

- PR #1733

File: roles/azuracast/upstream/docker-compose.sample.yml (hunk 1 above)
Comment (jvonau) [reply roles/azuracast/upstream/docker-compose.sample.yml:27]: What part of upstream and sample is unclear to you?

Comment (holta) [reply roles/azuracast/upstream/docker-compose.sample.yml:27]: Just to clarify, are all 4 files below effectively "documentation only" ?

```
root@box:/opt/iiab/iiab/roles/azuracast# find -name "*sample*"
./templates/azuracast.sample.env
./upstream/sample.env
./upstream/docker-compose.sample.yml
./upstream/azuracast.sample.env
```

Comment (jvonau) [reply roles/azuracast/upstream/docker-compose.sample.yml:27]: Documentation, upstream is current, the other is from a year ago see e7c85994d584894c9d40d827d308ae47e236078b

**Comment by holta** [comment]
Is this just about ready to merge?

What testing + guidance/documentation are most needed if so?

- PR #3353

**Comment by jvonau** [comment]
https://docs.azuracast.com/en/getting-started/installation/post-installation-steps

**Comment by holta** [comment]
> https://docs.azuracast.com/en/getting-started/installation/post-installation-steps

Great!

Can you draft an updated [azuracast/README.rst](https://github.com/iiab/iiab/tree/master/roles/azuracast#readme) ?

(Or README.md or README.adoc if that's preferred ?)

**Comment by holta** [comment]
> This PR was installed successfully onto an existing IIAB on a RPi 4 "183-lite64-SMALL+azuracast" here:
> 
> 10.8.0.46
> 
> @jvonau can you test/confirm that AzuraCast is working there as it should? (Hopefully 2GB RAM is enough for AzuraCast's Docker needs!)

10.8.0.46 being equivalent to 192.168.0.183 on this 64-bit Raspberry Pi OS on RPi 4 here:

Screenshot_20220831-071341

**Comment by holta** [comment]
Folks in North India want to test this, so let's make their lives easier by merging this.

Additional refinement PR's may be necessary **(so that instructions are crystal clear)** e.g. things like:

- PR #3353

**Pull request merged**